docs/schema/rapid7.md

```diff
@@ -0,0 +1,59 @@
+# Cartography - Rapid7 InsightVM Schema
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Table of contents
+
+- [Rapid7Host](#rapid7host)
+
+## Rapid7Host
+
+Placeholder representation of a single [Rapid7 Host or asset](https://help.rapid7.com/insightvm/en-us/api/index.html#operation/getAssets). This node is the minimal data necessary to map an asset.
+
+Warning! Work In Progress.
+
+
+| Field | Description |
+|-------|--------------|
+| firstseen| Timestamp of when a sync job first discovered this node  |
+| lastupdated |  Timestamp of the last time the node was updated |
+| r7_id | id |
+| r7_assessedForPolicies | assessedForPolicies |
+| r7_assessedForVulnerabilities | assessedForVulnerabilities |
+| hostname | hostName |
+| r7_ip | ip |
+| r7_mac | mac address |
+| r7_os | os |
+| r7_rawriskscore | rawRiskScore |
+| r7_riskscore | riskScore |
+| r7_type | type |
+| r7_addresses | addresses |
+| r7_configurations | configurations |
+| r7_databases | databases |
+| r7_files | files |
+| r7_history | history |
+| r7_hostnames | hostNames |
+| r7_ids | ids |
+| r7_links | links |
+| r7_osfingerprint | osFingerprint |
+| r7_services | services |
+| r7_software | software |
+| r7_usergroups | userGroups |
+| r7_users | users |
+| r7_vulnerabilities | vulnerabilities |
+
+### Relationships
+
+TBD
+* Azure Tenant contains one or more Rapid7 Hosts.
+```
+(AzureTenant)-[RESOURCE]->(Rapid7Host)
+```
+* Azure Subscription contains one or more Rapid7 Hosts.
+```
+(AzureSubscription)-[RESOURCE]->(Rapid7Host)
+```
+* Azure Virtual Machine is one single Rapid7 Host.
+* Similarly for other cloud providers and Onpremises.
```

Comment:
Still TBD? Your changes do not create these relationships.

Reply:
On relationships, I would be interested into some advices.
It does not seem most of the sources of cartography are doing cross-sources relations.
But this is an important point to me.
- link Cloud VM with tools present on it.
- problem, tool may have not always the information that makes relation certain. In case of Azure
* best case is to have resourceid (mde, rapid7 but does not seem always retrieved)
* crowdstrike gets only subscriptionid, resourcegroup and hostname
* worst case, only have hostname

I started to make AzureVM relation based on resourceid and where not available AzureSubscription based on subscriptionid.
I removed the latter as I found it making graph less readable.
I was thinking to use PRESENT_IN if relation based on resourceid and MAYBE_PRESENT_IN if relation based just on hostname.

What do you think?

Reply:
feedback on this?

Reply:
I don't have much context on how Azure works. But our preference is to only be certain about the relationships we create.

Reply:
the problem is always the same when dealing with multiple tools. is there a reliable key to use to join?
Azure resourceid could be but sadly not fetched reliably by most tools, at least in my context.
hostname/short_hostname is the default join key but hardly reliable (hostname reused, different case depending on tool, different length enforcement depending on tool - some enforce 15-chars windows limit...)

At this point, I'm plan to make relation base on resourceid or hostname/short_hostname.

Reply:
more to add here?